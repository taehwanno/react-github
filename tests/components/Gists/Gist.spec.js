import React from 'react';

import ReactGithub, { File, Gist } from '../../../src';

describe('<Gist />', () => {
  it('should match snapshot when render default', () => {
    expect(ReactGithub.render(<Gist />)).toMatchSnapshot();
  });

  it('should have description with props.description', () => {
    const description = 'Description of gist';
    const result = ReactGithub.render(<Gist description={description} />);
    expect(result.description).toBe(description);
  });

  it('should have public with props.isPublic', () => {
    const isPublic = false;
    const result = ReactGithub.render(<Gist isPublic={isPublic} />);
    expect(result.public).toBe(isPublic);
  });

  it('should render <File /> in children', () => {
    expect(ReactGithub.render((
      <Gist>
        <File name="file1.txt">contents of file1.txt</File>
        <File name="file2.txt">contents of file2.txt</File>
        <File name="file3.txt">contents of file3.txt</File>
      </Gist>
    ))).toMatchSnapshot();
  });
});
