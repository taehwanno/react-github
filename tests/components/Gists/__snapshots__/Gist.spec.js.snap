// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gist /> should match snapshot when render default 1`] = `
Object {
  "description": "",
  "files": Object {},
  "public": true,
}
`;

exports[`<Gist /> should render <File /> in children 1`] = `
Object {
  "description": "",
  "files": Object {
    "file1.txt": Object {
      "content": "contents of file1.txt",
    },
    "file2.txt": Object {
      "content": "contents of file2.txt",
    },
    "file3.txt": Object {
      "content": "contents of file3.txt",
    },
  },
  "public": true,
}
`;
